resource_cache/resource_cache.go

```diff
@@ -138,13 +138,22 @@ type ObjectCache struct {
 	config          *CacheConfig
 }
 
-func NewCacheConfig(scheme *runtime.Scheme, logKey interface{}, protectedGVKs map[schema.GroupVersionKind]bool, debugOptions DebugOptions) *CacheConfig {
+func NewCacheConfig(scheme *runtime.Scheme, possibleGVKs, protectedGVKs GVKMap, options ...Options) *CacheConfig {
```

Comment:
Why do we do this options array and then just pop the first entry off?

Reply:
because I wanted it to be optional